File: solutions/haskell/etl/1/src/ETL.hs
module ETL (transform) where

import qualified Data.Char as C
import qualified Data.Map as M

transform :: M.Map a String -> M.Map Char a
transform = M.fromList . foldl conv [] . M.toList
  where
    conv acc (k, v) =
      let ks = map C.toLower v
          vs = repeat k
       in zip ks vs ++ acc

File: solutions/go/etl/1/etl.go
package etl

import "strings"

func Transform(from map[int][]string) (to map[string]int) {
	to = make(map[string]int)
	for k, v := range from {
		for _, s := range v {
			to[strings.ToLower(s)] = k
		}
	}
	return
}
